src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -86,7 +86,15 @@ public Library GetLibrary(LibraryRange libraryRange, NuGetFramework targetFramew
             // This must exist in the external references
             if (_externalProjectsByUniqueName.TryGetValue(name, out ExternalProjectReference externalReference))
             {
-                packageSpec = externalReference.PackageSpec;
+                if (externalReference.PackageSpec == null ||
+                    libraryRange.VersionRange.FindBestMatch(new NuGetVersion[] { externalReference.PackageSpec.Version }) != null)
```

Comment:
`VersionRange` can be null to represent "all versions".
```suggestion
                    libraryRange.VersionRange?.FindBestMatch(new NuGetVersion[] { externalReference.PackageSpec.Version }) != null)
```

Reply:
I'm not sure how this should work yet. 
It's pretty surprising to me that we've gotten no null refs in the tests.